File: backend/Exceptions/Http/BaseHttpException.cs
using backend.Models;

namespace backend.Exceptions.Http;

public class BaseHttpException : BaseException
{
    public int Status { get; }
    public RequestVariant Variant { get; }

    public BaseHttpException(string message, int status, RequestVariant variant) : base(message)
    {
        this.Status = status;
        this.Variant = variant;
    }
}

File: web/src/exceptions/http/base-http-exception.ts
import { BaseExceptionProps, BaseException } from "../base-exception";
import { z } from "zod";

export const BaseHttpExceptionProps = BaseExceptionProps.extend({});

export class BaseHttpException extends BaseException {
  constructor({ message, variant }: z.infer<typeof BaseHttpExceptionProps>) {
    super({ message, variant });
  }
}